lib/spacesuit/auth_middleware.ex

```diff
@@ -4,21 +4,23 @@ defmodule Spacesuit.AuthMiddleware do
 
   @timed(key: "timed.authMiddleware-handle", units: :millisecond)
   def execute(req, env) do
-    session_service = Application.get_env(:spacesuit, :session_service)
-
     case req[:headers]["authorization"] do
       nil ->
         Logger.debug "No auth header"
         {:ok, req, env}
 
       "Bearer " <> token ->
-        case session_service[:enabled] do
-          true ->
-            session_service[:impl].handle_bearer_token(req, env, token, session_service[:url])
-          false -> 
+        with {true, :enabled?}      <- enabled?(),
+             {true, :handled_path?} <- handled_path?(req[:path]) do
+          session_service()[:impl].handle_bearer_token(req, env, token, session_service()[:url])
+        else
+          # the session service is disabled
+          {false, :enabled?} ->
+            Logger.debug "Session service disabled, skipping"
             {:ok, req, env}
-          _ ->
-            Logger.warn "Session service :enabled not configured!"
+
+          {false, :handled_path?} ->
+            Logger.warn "Session service bypassed for #{req[:path]}"
             {:ok, req, env}
```

Comment:
If it's reasonable to implement, we should validate it before passing it on. That way we don't pass junk inside.